### DISCUSSION TITLE
Should we find a new name for the `create` function?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/flow#9] [General]
### Context

In [v2-core](https://github.com/sablier-labs/v2-core) the create functions are called: 
- LockupLinear
  - `createWithDurations`
  - `createWithRange`
- LockupDynamic
  - `createWithDeltas`
  - `createWithMilestones`

In the `OpenEnded` conctract atm we have only one `create` function which is named simply "create".

Since additional logic for this contract will need to be added to the periphery, it's likely that we'll share the same periphery between `v2-core` and `v2-open-ended`. And we would have only one `Batch` contract.

Question: wouldn't it be confusing to have a function simply called `create` there? 

For context, the functions are called like this:
```solidity 
// for lockup
batch.createWithDurations(lockupLinear, asset, params);
batch.createWithRange(lockupLinear, asset, params);
batch.createWithDeltas(lockupDynamic, asset, params);
batch.createWithMilestones(lockupDynamic, asset, params);
// for open-ended
batch.create(openEnded, assset, params);
```

Atm, I don't have suggestions, will come later here if any good idea comes to mind

**Comment by IaroslavMazur** [comment]
> it's likely that we'll share the same periphery between `v2-core` and `v2-open-ended`. And we would have only one Batch contract.

What are the pros of doing things this way vs having the `v2-core` and `v2-open-ended` peripheries separate?

----

```
// for lockup
batch.createWithDurations(lockupLinear, asset, params);
batch.createWithRange(lockupLinear, asset, params);
batch.createWithDeltas(lockupDynamic, asset, params);
batch.createWithMilestones(lockupDynamic, asset, params);
// for open-ended
batch.create(openEnded, assset, params);
```

Given that there will be a single, joint periphery, the smallest viable change enough to suggestively distinguish between the `v2-core` and `v2-open-ended` create functions would be to transform `batch.create(openEnded, assset, params);` into smth like `batch.createOpenEnded(openEnded, assset, params);`

However, I'd, rather, suggest incorporating some sort of organization into the function naming, such that each function clearly references the Stream type it's capable of creating (`v2-core` or `v2-open-ended`).

---

Side note: should we, maybe, refer to the `v2-core` Streams as `Lockup` or `close-ended`, to contrast the `open-ended` ones?

**Reply by andreivladbrg** [reply]
> What are the pros of doing things this way vs having the v2-core and v2-open-ended peripheries separate?

Need to think about it, might be better to add in this project a periphery dir inside `src`

**Comment by smol-ninja** [comment]
Given that,
1. We have standardised `create` function names in lockup contracts with `createWithDurations` and `createWithTimestamps`
2. We have [decided to keep OE periphery](https://github.com/sablier-labs/v2-open-ended/issues/10) in the OE repo itself

This discussion is no longer required, and I do not see a problem with `create` anymore. cc @andreivladbrg what do you think?

**Comment by andreivladbrg** [comment]
Yes, you are right, this discussion is no longer required
